File: 프로그래머스/lv2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let P = priorities.map((v,i)=> [i,v]);
    let max = priorities.sort((a,b)=>b-a);
    let answer = 0;
    console.log(P)
    while(true){
        if(P[0][1] === max[0]){
            if(P[0][0] === location){
                return ++answer;
            }else{
                P.shift();
                max.shift();
                answer++;
            }
        }else{
            let temp = P.shift();
            P.push(temp);
        }
    }
    return answer;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.Comparator;
import java.util.PriorityQueue;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;

        PriorityQueue<Integer> ps = new PriorityQueue<>(Comparator.reverseOrder());

        for (int priority : priorities) {
            ps.add(priority);
        }

        System.out.println("ps = " + ps);

        while (!ps.isEmpty()) {
            for (int i = 0; i < priorities.length; i++) {
                if (priorities[i] == ps.peek()) {
                    ps.poll();
                    answer++;

                    if (location == i) {
                        return answer;
                    }
                }
            }
        }

        return answer;
    }
}